### DISCUSSION TITLE
The client does not connect to the server (if it has been unavailable for a while)

### DISCUSSION THREAD

**Discussion by farmatra** [rustdesk/rustdesk#7115] [General]
### Bug Description

Client version 1.2.4, build from: 2024-02-07. After some time without access to the server, the client stops trying to connect to the server. The client's connection status is "Not connected. Check the connection.". On versions 1.2.4 built before January, this was not noticed. After restarting the client part, the connection is restored.

### How to Reproduce

Block access to the RasterDesk server for a while.

### Expected Behavior

Reconnecting the client to the server after restoring access to the server.

### Operating system(s) on local side and remote side

Windows 10

### RustDesk Version(s) on local side and remote side

1.2.4

### Screenshots

![204830](https://github.com/rustdesk/rustdesk/assets/16952400/8dd67dcd-b951-41d5-a0c6-63f57a2b9ed3)

### Additional Context

Connecting the Client to the server after the server has become available.

**Comment by rustdesk** [comment]
Before, it reconnect every 1 second during network failure, now it reconnect every 32 seconds.

**Comment by zhangbo8418** [comment]
> Before, it reconnect every 1 second during network failure, now it reconnect every 32 seconds.

https://github.com/rustdesk/rustdesk/discussions/7045
Same problem as mine, it won't reconnect, even if I wait 6 hours. . .

**Reply by rustdesk** [reply]
Mine can

**Reply by rustdesk** [reply]
I suggest you `cargo run` to get some clue since you can easily reproduce and know which part of related code.

**Comment by fufesou** [comment]
Maybe the same reason https://github.com/rustdesk/rustdesk/pull/7162

@farmatra Hi, did you have your PC slept?

I can only reproduce this issue when my computer sleeps for at least 20 minutes.

**Comment by zhangbo8418** [comment]
> Maybe the same reason #7162
> 
> @farmatra Hi, did you have your PC slept?
> 
> I can only reproduce this issue when my computer sleeps for at least 20 minutes.

PC and Mac will encounter this issue after being in sleep mode for a long time (but this is not normal! Reconnecting should not require restarting the RustDesk service). Android is set to start up automatically. If the DDNS IP changes, connection will also be impossible without opening the RustDesk application.

**Reply by rustdesk** [reply]
As said previously, we can not reproduce, you'd better `cargo run` to get some clue. I can not simply revert our code change because of an issue we can not reproduce.
